pkg/apis/core/v1alpha1/defaults_test.go

```diff
@@ -364,6 +364,12 @@ var _ = Describe("Defaults", func() {
 			Expect(obj.Spec.Settings.VerticalPodAutoscaler.Enabled).To(Equal(vpaEnabled))
 			Expect(obj.Spec.Settings.OwnerChecks.Enabled).To(Equal(ownerChecks))
 		})
+
+		It("should default ipFamilies setting to IPv4 single-stack", func() {
```

Comment:
Is there a reason testing the defaulting via the parent func (`SetObjectDefaults_Seed`) and not by invoking directly the dedicated func `SetDefaults_SeedNetworks`? At the end, both of the things will give the same result but for consistency reasons we should strictly follow only 1 of these 2 approaches.

Reply:
This relates to my reasoning described here: https://github.com/gardener/gardener/pull/7288#discussion_r1065730015

From my point of view, all unit tests should call the actual defaulting func (as done in this test case) because that's the actual unit that will be used in the production code.
Only testing `SetDefaults_SeedNetworks` would not cover the entire "truth". These tests can pass even if the actual overall defaulting logic doesn't set the networks section.

> for consistency reasons we should strictly follow only 1 of these 2 approaches

You're right, let's make this consistent everywhere.
If you agree with my point above, we can set this straight and call the actual overall defaulting logic in all unit tests.
I could do it in a dedicated PR or as a new commit in this PR – whatever you prefer :)

Reply:
I started preparing this tech debt cleanup.
If we already touch the defaulting code and unit tests, we should perform all of the following changes:

- extract defaulting code and unit tests into dedicated files (like the dedicated `types_*.go` files)
- refactoring the defaulting code to `SetDefaults_*` functions (as they are supposed to be structured)
- testing the overall defaulting function (mentioned above)

Right now, our defaulting code and unit tests are a bit messy in these aspects and we could set things straight now.
I performed the changes for the `v1beta1.Project` defaulting as an example: https://github.com/stackitcloud/gardener/compare/ipv6-api...timebertt:gardener:tech-debt/defaults
These steps would need to be performed for all API groups, versions, and kinds.

Doing so in this PR doesn't seem right and will distract the review. 
If you like my example and agree with my approach, I will open an issue for this and perform the changes in a dedicated commit :)

Reply:
Sounds reasonable, fine for me 👍

Reply:
Great, opened an issue for this: https://github.com/gardener/gardener/issues/7312